### PR TITLE
Explicit list of Souffle tests

### PR DESCRIPTION
**Pull request by mbudiu-vmw**

### CODE REVIEW

**Comment by ryzhyk** [comment]
This PR, along with #346, is still failing.  This appears to be due to a combination of a missing dependency on MacOS and new/changed souffle tests. I am going to close these PRs, merge them into my local branch and eventually create a fixed PR (probably tomorrow).

File: .travis.yml [hunk 1]
```diff
@@ -81,7 +81,7 @@ script:
 #- DDLOG_TEST_PROGRESS=1 stack --no-terminal test --haddock --no-haddock-deps
 #- DDLOG_TEST_PROGRESS=1 stack --no-terminal test --ta "$TEST_SUITE"  --haddock --no-haddock-deps
 - if [ x${TOOL} = xstack ]; then DDLOG_TEST_PROGRESS=1 stack --no-terminal test --ta "$TEST_SUITE"; fi
-- if [ x${TOOL} = xrun_souffle_tests ]; then cd test && travis_wait 30 ./run-souffle-tests.py -s 0 -r 17; fi
+- if [ x${TOOL} = xrun_souffle_tests ]; then cd test && travis_wait 30 ./run-souffle-tests.py  ackermann andersen bigrams cba_expr_value centroids clique cliquer comp-parametrized-inherit comp-parametrized-multilvl degree dfa dfa_min disconnected dominance double_tree earley equal factoring family fib; fi
```
[the remark this answers is not in the document]
Comment (mbudiu-vmw) [reply .travis.yml]: I don't know the yaml syntax to break a line into multiple lines. I will have to investigate.

**Comment by mbudiu-vmw** [comment]
Ideally we should always run all souffle tests, but the Rust compiler croaks if you do that.
That's why we added the option to run them in batches. This line only runs 20 tests, more than the previous 17 we used to. We should also run all batches, but that takes too long.

**Comment by ryzhyk** [comment]
> I am fine with the change, though it appears to me that we do reduce the number of tests run. Would it make sense to document that (e.g., by mentioning the set of tests run before and after your change in the change description).
> 
> Also, with this PR merged, do we will need the `-r/--run` and `-s/--skip` options to `run-souffle-tests.py`? Perhaps I am misunderstanding but it seems inherently brittle to just skip a certain number of tests because if new tests are now added to the front of the list upstream, we will suddenly run a different set of tests, won't we? To me it would seem preferable to get rid of the `svn` logic inside of that script altogether, but let me know if I am missing something.

We never ran all tests in Travis, as it would take way too long and we'd have to break them up into many Travis test runs. The PR just uses symbolic names instead of the range we used in the past.

Pulling tests from SVN is generally good, as this way we will be able to discover new bugs in the Souffle converter as the test suite grows. I currently run the full Souffle test suite by hand before every release. We'll start doing this automatically once we've moved to a more powerful test platform.  @mbudiu-vmw maintains a blacklist of tests that are currently not supported due to known bugs.

**Pull request merged**